src/utils/calendar.ts

```diff
@@ -0,0 +1,34 @@
+import { PersonalizedVenueEvent } from "types/venues";
+import { VenueEvent } from "types/venues";
+import { eventEndTime, eventStartTime } from "utils/event";
+import { WithVenueId } from "utils/id";
+import { getFullVenueInsideUrl } from "utils/url";
+
+import ical from "ical-generator";
+
+export const createCalendar = (
+  allEvents: PersonalizedVenueEvent[] | WithVenueId<VenueEvent>[],
+  calendarname: string
+) => {
+  const cal = ical({ name: calendarname });
+
+  if (!cal) return;
+
+  allEvents.map((event: PersonalizedVenueEvent | WithVenueId<VenueEvent>) =>
+    cal.createEvent({
+      id: event.venueId,
+      start: eventStartTime(event),
+      end: eventEndTime(event),
+      description: event.host,
+      summary: event.name ? event.host : "",
```

Comment:
What's the intent of this ternary here? Currently it will check if `event.name` is truthy, and if it is, use `event.host`. If `event.name` is falsy it will use `""`.

I expect what we want here is to use event.name if present, if not fall back to event.host, and if not that then fall back to the empty string (or similar)?

If so, that would be implemented using TypeScript's "default the value if empty" `??` (nullish coalescing) operator as follows:

```suggestion
      summary: event.name ?? event.host ?? "",
```

Though looking at the definition of `VenueEvent` in `src/types/venues.ts`, `name`/`host` can never be `undefined` anyway.. so not sure what the intent of this is, we can probably just be as simple as this here:

```suggestion
      summary: event.name,
```

Further Reading:

- https://www.typescriptlang.org/docs/handbook/release-notes/typescript-3-7.html#nullish-coalescing
- https://sebbo2002.github.io/ical-generator/develop/reference/classes/icalevent.html#summary

Reply:
fixed in https://github.com/sparkletown/sparkle/pull/1423/commits/ec68e834fab42e9549a285d7a454e4c29e1f24e4